pin-project-internal/src/pin_project/enums.rs

```diff
@@ -86,18 +91,27 @@ fn named(
     let mut proj_body = Vec::with_capacity(fields.len());
     let mut proj_fields = Vec::with_capacity(fields.len());
     for Field { attrs, ident, ty, .. } in fields {
-        if let Some(attr) = attrs.find_remove(PIN) {
-            let _: Nothing = syn::parse2(attr.tokens)?;
-            cx.push_unpin_bounds(ty.clone());
+        let cfg = collect_cfg(attrs);
+        if cx.find_pin_attr(attrs)? {
             let lifetime = &cx.lifetime;
-            proj_fields.push(quote!(#ident: ::core::pin::Pin<&#lifetime mut #ty>));
-            proj_body.push(quote!(#ident: ::core::pin::Pin::new_unchecked(#ident)));
+            proj_fields.push(quote! {
+                #(#cfg)* #ident: ::core::pin::Pin<&#lifetime mut #ty>
```

Comment:
(In the future, it may be preferable to move these processes to proc-macro-derive.)